### ISSUE TITLE
Are the Result<T, Error> type definitions wrong?

### ISSUE THREAD

**Issue by tobias-tengler** [facebook/relay#4819]
I wanted to play around with `@catch` and noticed that the Flow / TypeScript types and the compiler generated errors passed to the Result type don't make much sense.

The compiler will generate the following code:

```flow
export type FooQuery$data = {|
  +me: Result<?{|
    +firstName: ?string,
    +lastName: ?string,
  |}, $ReadOnlyArray<mixed>>,
|};
```

so the error type is `$ReadOnlyArray<mixed>`.

The type definitions in relay-runtime define the errors field on the `ErrorResult` as a `$ReadOnlyArray<Error>`:
https://github.com/facebook/relay/blob/e357063c26d0aaf0a5027ba591138ad172cd103f/packages/relay-runtime/experimental.js#L46C2-L55C67

So you'd end up with the `ErrorResult.errors` being `Array<Array<Error>>`. This doesn't seem right to me?

**Comment by captbaritone** [comment]
Maybe this is a bug where we've used a global type `Error` as the name of a type generic?

**Comment by itamark** [comment]
@captbaritone yup

<img width="319" alt="Screenshot 2024-10-15 at 12 19 18" src="https://github.com/user-attachments/assets/c7ededcc-cddf-4731-bcc4-f5ae1bf28013">

**Comment by captbaritone** [comment]
I see, I missed the point. `Array<Array<Error>>`. Yeah, looks like a bug in the types.

**Comment by tobias-tengler** [comment]
Fixed by 9571d80e5cecfb052858c9fa11cc9dea9780318e
